api/lib/infrastructure/repositories/campaign/target-profile-for-specifier-repository.js

```diff
@@ -16,49 +15,59 @@ function _fetchTargetProfiles(organizationId) {
     .select('targetProfileId')
     .from('target-profile-shares')
     .where({ organizationId });
-  return (
-    knex('target-profiles')
-      .select([
-        'target-profiles.id',
-        'target-profiles.name',
-        'target-profiles.description',
-        'target-profiles.category',
-        knex.raw('ARRAY_AGG("skillId") AS "skillIds"'),
-        knex.raw('ARRAY_AGG("badges"."id")  AS "badgeIds"'),
-        knex.raw('ARRAY_AGG("stages"."id")  AS "stageIds"'),
-        knex.raw('ARRAY_AGG("target-profile_tubes"."tubeId")  AS "tubeIds"'),
-      ])
+  return knex('target-profiles')
+    .select([
+      'target-profiles.id',
+      'target-profiles.name',
+      'target-profiles.description',
+      'target-profiles.category',
       // TODO remove it after target profile tube migration to target-profile_tubes
-      .leftJoin('target-profiles_skills', 'target-profiles_skills.targetProfileId', 'target-profiles.id')
-      .leftJoin('target-profile_tubes', 'target-profile_tubes.targetProfileId', 'target-profiles.id')
-      .leftJoin('badges', 'badges.targetProfileId', 'target-profiles.id')
-      .leftJoin('stages', 'stages.targetProfileId', 'target-profiles.id')
-      .where({ outdated: false })
-      .where((qb) => {
-        qb.orWhere({ isPublic: true });
-        qb.orWhere({ ownerOrganizationId: organizationId });
-        qb.orWhereIn('target-profiles.id', selectTargetProfileSharesIdsBelongToOrganization);
-      })
-      .groupBy('target-profiles.id')
-  );
+      knex
+        .select(knex.raw('ARRAY_AGG("skillId")'))
+        .from('target-profiles_skills')
+        .where(knex.raw('"target-profiles_skills"."targetProfileId"="target-profiles".id'))
+        .as('skillIds'),
+      knex
+        .count(knex.raw('"badges"."id"'))
```

Comment:
Ça se tente sans le `raw` ?
```suggestion
        .count('id')
```